fe/topology.py

```diff
@@ -390,6 +555,163 @@ def parameterize_improper_torsion(self, ff_params):
+
+        return combined_qlj_params, nb_potential.interpolate()
+
+
+def get_ring_membership(mol):
+    """
+    Get the membership of each atom in the mol. The algorithm
+    finds islands that are formed connected by bond bridges.
+
+    Let N = mol.GetNumAtoms()
+
+    The returned array is of length N, where each atom is marked
+    by a membership class K such that 0 < K < N.
+    """
+
+    g = networkx.Graph()
+    for bond in mol.GetBonds():
+        g.add_edge(bond.GetBeginAtomIdx(), bond.GetEndAtomIdx())
+
+    # find bridges and remove them to form islands
+    for bridge in networkx.bridges(g):
+        g.remove_edge(*bridge)
+
+    # find connected components after bridge removal
+    membership = np.zeros(mol.GetNumAtoms(), dtype=np.int32)
+    for group_idx, atom_list in enumerate(networkx.connected_components(g)):
+        for atom_idx in atom_list:
+            membership[atom_idx] = group_idx
+
+    return membership
+
+
+class DualTopologyStandardDecoupling(DualTopology):
+    """
+    Standardized variant, where both ligands A and B have their charges, sigma, and epsilons are set
+    to standard, forcefield-independent values. There is no parameter interpolation. lambda=0 has both
+    Ligand A and B fully in the pocket. lambda=1 has ligand b fully decoupled, while ligand a is kept
+    in place.
+
+    Furthermore, the ligand's non-ring torsions are turned off at the standard state to improve
+    sampling.
+
+    """
+    def parameterize_proper_torsion(self, ff_params):
+        # (ytz): TBD need to do this for Base Topology as well
+
+        torsion_params, torsion_potential = super().parameterize_proper_torsion(ff_params)
+
+        mol_c = Chem.CombineMols(self.mol_a, self.mol_b)
+        membership = get_ring_membership(mol_c)
+
+        num_torsions = torsion_params.shape[0]
+
+        lambda_mult_idxs = np.zeros(num_torsions, dtype=np.int32)
+        lambda_offset_idxs = np.ones(num_torsions, dtype=np.int32)
+
+        for torsion_idx, (_, b, c, _) in enumerate(torsion_potential.get_idxs()):
+            if membership[b] != membership[c]:
+                lambda_offset_idxs[torsion_idx] = 0
+
+        torsion_potential.set_lambda_mult_and_offset(lambda_mult_idxs, lambda_offset_idxs)
+
+        return torsion_params, torsion_potential
+
+    def parameterize_nonbonded(self,
+        ff_q_params,
+        ff_lj_params):
+
+        # both mol_a and mol_b are standardized.
+        # we don't actually need derivatives for this stage.
+
+        # zero out/stop grad
+        ff_q_params = np.zeros_like(ff_q_params)
+        ff_lj_params = np.zeros_like(ff_lj_params)
+
+        qlj_params, nb_potential = super().parameterize_nonbonded(ff_q_params, ff_lj_params)
+        qlj_params = jax.ops.index_update(qlj_params, jax.ops.index[:, 0], _STANDARD_CHARGE)
+        mol_c = Chem.CombineMols(self.mol_a, self.mol_b)
+        sig_eps = simple_lj_typer(mol_c)
```

Comment:
Similar comment as above -- is there a way to expose this choice in either the `parameterize_nonbonded` function signature, or in the `DualTopologyStandardDecoupling` constructor?

Reply:
I'm actually not sure I want to expose this to the end-user yet. Part of this is because we need to take care that the user supplies the same simple_qlj_typer in all three classes (you're not really allowed to mix and match them in order for the end states to be connected).